src/object/cli_obj.c

```diff
@@ -2393,16 +2404,66 @@ obj_update_csums(const struct dc_object *obj, const daos_obj_update_t *args) {
 	}
 }
 
+/* Selects next replica in the object's layout.
+ */
+static int
+obj_retry_csum_err(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
+		     uint64_t dkey_hash, unsigned int map_ver, uint8_t *bitmap)
+{
+	struct daos_oclass_attr	*oca;
+	unsigned int next_shard, retry_size, shard_cnt, shard_idx;
+	int rc = 0;
+
+	/* CSUM retry for EC object is not supported yet */
+	if (daos_oclass_is_ec(obj->cob_md.omd_id, &oca)) {
```

Comment:
looks better to avoid call daos_oclass_is_ec for another time, as the daos_oclass_attr_find() will add some overhead. probably can do it only once in a IO request. So after we do it the first time can set a flag to avoid do it again.  Or cache the oca in obj_auxi